src/core_init_atmosphere/mpas_init_atm_static.F

```diff
@@ -829,107 +831,146 @@ subroutine init_atm_static(mesh, dims, configs)
-             lat_pt = lat_pt * PI / 180.0
-             lon_pt = lon_pt * PI / 180.0
+            if (tile % is_processed) then
+                cycle
+            end if
 
-             iPoint = nearest_cell(lat_pt,lon_pt,iPoint,nCells,maxEdges, &
-                                   nEdgesOnCell,cellsOnCell, &
-                                   latCell,lonCell)
-             if (rarray(ii,jj,1) /= msgval) then
+            call mpas_log_write('Processing tile: '//trim(tile % fname))
+
+            all_pixels_mapped_to_halo_cells = .true.
+
+            do j = supersample_fac * tile_bdr + 1, supersample_fac * (tile_ny + tile_bdr), 1
+                do i = supersample_fac * tile_bdr + 1, supersample_fac * (tile_nx + tile_bdr), 1
+
+                    ii = (i - 1) / supersample_fac + 1
+                    jj = (j - 1) / supersample_fac + 1
+
+                    sval = tile % tile(ii, jj, 1)
+
+                    call mgr % tile_to_latlon(tile, j, i, lat_pt, lon_pt, supersample_fac)
+                    call mpas_latlon_to_xyz(xPixel, yPixel, zPixel, sphere_radius, lat_pt, lon_pt)
+                    call mpas_kd_search(tree, (/xPixel, yPixel, zPixel/), res)
+
+                    if (tile % tile(ii, jj, 1) * scalefactor /= missing_value) then
+                        if (bdyMaskCell(res % id) < nBdyLayers) then
+                            !
+                            ! This field only matters for land cells, and for all but the outermost boundary cells,
+                            ! we can safely assume that the nearest model grid cell contains the pixel (else, a different
+                            ! cell would be nearest)
+                            !
+                            if (landmask(res % id) == 1) then
+                                snoalb_integer(res % id) = snoalb_integer(res % id) + int(sval, kind=I8KIND)
+                                nhs(res % id) = nhs(res % id) + 1
+                            end if
+
+                            if (res % id <= nCellsSolve) then
+                                all_pixels_mapped_to_halo_cells = .false.
+                            end if
+                        ! For outermost land cells, additional work is needed to verify that the pixel
+                        ! actually lies within the nearest cell
+                        else
+                            if (mpas_in_cell(xPixel, yPixel, zPixel, xCell(res % id), yCell(res % id), zCell(res % id), &
+                                              nEdgesOnCell(res % id), verticesOnCell(:,res % id), xVertex, yVertex, zVertex)) then
+
+                                if (landMask(res % id) == 1) then
+                                    snoalb_integer(res % id) = snoalb_integer(res % id) + int(sval, kind=I8KIND)
+                                    nhs(res % id) = nhs(res % id) + 1
+                                end if
+
+                                if (res % id <= nCellsSolve) then
```

Comment:
Would the code at line 923 obviate the need for the same code here?

Reply:
Yes, looks like I forgot to remove the code at line 932.